x/launch/types/genesis.go

```diff
@@ -27,6 +28,16 @@ func (gs GenesisState) Validate() error {
 	// this line is used by starport scaffolding # ibc/genesistype/validate
 
 	// this line is used by starport scaffolding # genesis/types/validate
+	// Check for duplicated index in chainNameCount
+	chainNameCountIndexMap := make(map[string]struct{})
+
+	for _, elem := range gs.ChainNameCountList {
+		index := string(ChainNameCountKey(elem.ChainName))
+		if _, ok := chainNameCountIndexMap[index]; ok {
+			return fmt.Errorf("duplicated index for chainNameCount")
+		}
+		chainNameCountIndexMap[index] = struct{}{}
+	}
```

Comment:
maybe we should check if the `ChainNameCountList` object exists in the `ChainlList`

Reply:
Yes that's a good point.
By the way we should in fact also verify that this count is above all the number of the chain id that contains this name but to do this verification properly we should refactor `chainID` from type `string` into `{chainName: string, number: uint64}`
What we can do for now is to add the check you mentioned and replace the current critical error we have for already existing chainID with a simple error

Reply:
Thanks for spotting this. I'm also adding a check to verify the count is above all chain ID name and check chain ID is correct